app/Models/BuildPropagationHistory.php

```diff
@@ -30,4 +32,42 @@ class BuildPropagationHistory extends Model
     protected $dates = [
         'created_at',
     ];
+
+    public function build()
+    {
+        return $this->belongsTo(Build::class, 'build_id');
+    }
+
+    public function scopeBaseChangelog($query, $streamId, $days)
+    {
+        $buildsTable = with(new Build)->getTable();
+        $propagationTable = with(new self)->getTable();
+        $streamsTable = 'osu_updates.'.with(new UpdateStream)->getTable();
+
+        $query->join($buildsTable, "{$buildsTable}.build_id", '=', "{$propagationTable}.build_id")
+            ->join($streamsTable, "{$streamsTable}.stream_id", '=', "{$buildsTable}.stream_id")
+            ->select(DB::raw('created_at, cast(sum(user_count) as signed) as user_count'))
+            ->where('created_at', '>', Carbon::now()->subDays($days));
+
+        if ($streamId !== null) {
+            $query->where("{$buildsTable}.stream_id", $streamId);
+        }
+    }
+
+    public function scopeChangelog($query, $streamId, $days)
+    {
+        $buildsTable = with(new Build)->getTable();
+
+        $query->baseChangelog($streamId, $days);
+
+        if ($streamId !== null) {
+            $query->addSelect(DB::raw("{$buildsTable}.version as label"))
```

Comment:
not even need to group by for build history (or joining with streams table at all)